### DISCUSSION TITLE
Remove synchronized works to make it Virtual threads(Loom) friendly

### DISCUSSION THREAD

**Discussion by nitin1532** [aws/amazon-redshift-jdbc-driver#108] [General]
There are lot of synchronized keyword, and its getting it Pinned when used with virtual threads enabled. Can you please take the enhancement request to make it Loom friendly in next release.
Currently I am seeing multiple onPinned() events for QueryExecutorImpl.java and method closeStatementAndPortal(). but i could see a lot more places it needs to be removed.

Thanks
Nitin Singhal

**Comment by bhvkshah** [comment]
Hi @nitin1532 there are several high priority tasks the team is working on currently,. I will add this enhancement request to our backlog but I cannot yet give an ETA. Of course, you are welcome to submit a Pull Request which I will be happy to review.

**Comment by nitin1532** [comment]
Can you guide me the process for it? Thanks.

Regards,
Nitin Singhal

**Comment by bhvkshah** [comment]
Gladly, you can follow the guide here:https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project on creating your own branch by forking from the master branch, committing your code changes, and submitting a PR for your changes.
